### PR TITLE
Side staking tests v4

### LINKED ISSUE #548: Implement unit tests for Side Staking in V4
Implement unit tests for Side Staking in V4 that can be found here: https://github.com/oceanprotocol/contracts/blob/v4main/test/flow/SideStaking.test.js

### PR DESCRIPTION
**Pull request by maldo-ctrl**
Fixes #548 .

Changes proposed in this PR:
- Add Side Staking tests for v4
- Added back `cap` argument in `deploy_erc721_erc20` helper function: https://github.com/oceanprotocol/ocean.py/pull/573#discussion_r746405653

image

### CODE REVIEW

**Review by mariacarmina — APPROVED** [review]
Nice work, @maldo-ctrl! Looks good from my side.

File: ocean_lib/models/v4/test/test_side_staking.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+    initial_ocean_liquidity = web3.toWei("10", "ether")
+
+    ocean_token = ERC20Token(web3, get_address_of_type(config, "Ocean"))
+
+    # Deploy erc721 and erc20 data token
+    erc721, erc20 = deploy_erc721_erc20(
+        web3, config, consumer_wallet, consumer_wallet, web3.toWei("10000", "ether")
+    )
+
+    if ocean_token.balanceOf(consumer_wallet.address) == 0:
```
Comment (calina-c) [inline ocean_lib/models/v4/test/test_side_staking.py:39]: can we move these in some helpers? I get the feeling this is a common case: if balance falls, send some OCEAN.

**Review by calina-c — CHANGES_REQUESTED** [review]
just a small suggestion

File: tests/resources/helper_functions.py [hunk 2]
```diff
@@ -257,7 +257,7 @@ def deploy_erc721_erc20(
             erc721_publisher.address,
             ZERO_ADDRESS,
         ],
-        uints=[web3.toWei("0.5", "ether"), 0],
+        uints=[cap, 0],
```
Comment (maldo-ctrl) [inline tests/resources/helper_functions.py:260]: Cap argument added as it was accidentally removed here: https://github.com/oceanprotocol/ocean.py/commit/f28d07c0300f2f4810f193708531e78004eab0c1

File: tests/resources/helper_functions.py [hunk 3]
```diff
@@ -217,11 +217,11 @@ def deploy_erc721_erc20(
     config: Config,
     erc721_publisher: Wallet,
     erc20_minter: Optional[Wallet] = None,
+    cap=Web3.toWei("0.5", "ether"),
```
Comment (maldo-ctrl) [inline tests/resources/helper_functions.py:220]: Cap argument added as it was accidentally removed here: https://github.com/oceanprotocol/ocean.py/commit/f28d07c0300f2f4810f193708531e78004eab0c1

**Review by calina-c — APPROVED** [review]

**Pull request merged**